As a user, if I include sampling uncertainty in an evaluation of freezing level with precip as a covariate, I don't expect a NoSuchElementException

There is probably a good reason for the failure, but I'm guessing a NoSuchElementException is not the way to report it.  Here is the exception:

```
wres.pipeline.InternalWresException: Could not complete project execution
	at wres.pipeline.Evaluator.evaluate(Evaluator.java:406)
	at wres.pipeline.Evaluator.evaluate(Evaluator.java:178)
	at wres.Functions.evaluate(Functions.java:135)
	at wres.server.EvaluationService.lambda$startEvaluation$2(EvaluationService.java:698)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: wres.pipeline.WresProcessingException: Encountered an error while processing evaluation 'ViZVhA45aN4yZzODq0s1wRfu2ZQ': 
	at wres.pipeline.Evaluator.evaluate(Evaluator.java:827)
	at wres.pipeline.Evaluator.evaluate(Evaluator.java:377)
	... 7 common frames omitted
Caused by: java.util.concurrent.CompletionException: java.util.NoSuchElementException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
	... 3 common frames omitted
Caused by: java.util.NoSuchElementException: null
	at java.base/java.util.TreeMap.key(TreeMap.java:1602)
	at java.base/java.util.TreeMap.firstKey(TreeMap.java:291)
	at java.base/java.util.TreeSet.first(TreeSet.java:391)
	at java.base/java.util.Collections$UnmodifiableSortedSet.first(Collections.java:1201)
	at wres.datamodel.bootstrap.BootstrapPool.lambda$new$0(BootstrapPool.java:215)
	at java.base/java.util.Comparator.lambda$comparing$77a9974f$1(Comparator.java:473)
	at java.base/java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
	at java.base/java.util.TimSort.sort(TimSort.java:234)
	at java.base/java.util.Arrays.sort(Arrays.java:1307)
	at java.base/java.util.ArrayList.sort(ArrayList.java:1721)
	at wres.datamodel.bootstrap.BootstrapPool.<init>(BootstrapPool.java:222)
	at wres.datamodel.bootstrap.BootstrapPool.of(BootstrapPool.java:66)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.<init>(StationaryBootstrapResampler.java:851)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.of(StationaryBootstrapResampler.java:125)
	at wres.pipeline.pooling.PoolProcessor.getSamplingUncertaintyStatistics(PoolProcessor.java:579)
	at wres.pipeline.pooling.PoolProcessor.createStatistics(PoolProcessor.java:492)
	at wres.pipeline.pooling.PoolProcessor.get(PoolProcessor.java:151)
	at wres.pipeline.pooling.PoolProcessor.get(PoolProcessor.java:60)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	... 3 common frames omitted
```

The declaration is below. Its the freezing level evaluation used to illustrate covariates for the upcoming QPR with sampling uncertainty added.  The data to reproduce should already be available since that was the same evaluation used to test covariates.

Thanks,

Hank

====================

```yml
label: QPR eval with covariates
observed:
  sources: .../WWBC1HUF.ZELV
  time_zone_offset: -6
predicted:
  sources: .../WWBC1HUF.zelv.hindcasts.tgz
covariates:
  - sources: .../WWBC1HUF.MRGHO.MAP06
    minimum: 0.01
    time_zone_offset: -6
features:
  - {observed: WWBC1HUF, predicted: WWBC1HUF}
lead_times:
  minimum: 0
  maximum: 368
  unit: hours
lead_time_pools:
  period: 6
  unit: hours
season:
  minimum_day: 1
  minimum_month: 12
  maximum_day: 31
  maximum_month: 3
probability_thresholds:
  values: [0.25]
  operator: greater equal
  apply_to: observed
sampling_uncertainty:
  sample_size: 500
  quantiles: [0.05,0.95]
duration_format: hours
decimal_format: '#0.000000'
output_formats:
  - csv2
  - pairs
  - png
```

Reproduced:

Caused by: java.util.NoSuchElementException: null
	at java.base/java.util.TreeMap.key(TreeMap.java:1602)
	at java.base/java.util.TreeMap.firstKey(TreeMap.java:291)
	at java.base/java.util.TreeSet.first(TreeSet.java:391)
	at java.base/java.util.Collections$UnmodifiableSortedSet.first(Collections.java:1201)
	at wres.datamodel.bootstrap.BootstrapPool.lambda$new$0(BootstrapPool.java:215)
	at java.base/java.util.Comparator.lambda$comparing$77a9974f$1(Comparator.java:473)
	at java.base/java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
	at java.base/java.util.TimSort.sort(TimSort.java:234)
	at java.base/java.util.Arrays.sort(Arrays.java:1307)
	at java.base/java.util.ArrayList.sort(ArrayList.java:1721)
	at wres.datamodel.bootstrap.BootstrapPool.<init>(BootstrapPool.java:222)
	at wres.datamodel.bootstrap.BootstrapPool.of(BootstrapPool.java:66)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.<init>(StationaryBootstrapResampler.java:851)
	at wres.datamodel.bootstrap.StationaryBootstrapResampler.of(StationaryBootstrapResampler.java:125)
	at wres.pipeline.pooling.PoolProcessor.getSamplingUncertaintyStatistics(PoolProcessor.java:579)
	at wres.pipeline.pooling.PoolProcessor.createStatistics(PoolProcessor.java:492)
	at wres.pipeline.pooling.PoolProcessor.get(PoolProcessor.java:151)
	at wres.pipeline.pooling.PoolProcessor.get(PoolProcessor.java:60)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)

--- a/wres-datamodel/src/wres/datamodel/bootstrap/BootstrapPool.java
+++ b/wres-datamodel/src/wres/datamodel/bootstrap/BootstrapPool.java
@@ -128,56 +128,97 @@ class BootstrapPool<T>
         {
             if ( next.size() > 1 )
             {
-                // Offsets between consecutive series
-                for ( int i = 1; i < next.size(); i++ )
-                {
-                    Instant first;
-                    List<Event<T>> firstSeries = next.get( i - 1 );
+                Set<Duration> nextOffsets = this.getOffsetsBetweenConsecutiveSeries( next );
+                offsets.addAll( nextOffsets );
+            }
+        }
 
-                    if ( this.hasForecasts() )
-                    {
-                        first = firstSeries.get( 0 )
-                                           .getTime();
-                    }
-                    else
-                    {
-                        first = firstSeries.get( firstSeries.size() - 1 )
-                                           .getTime();
-                    }
+        return Collections.unmodifiableSet( offsets );
+    }
 
-                    Instant last = next.get( i )
-                                       .get( 0 )
-                                       .getTime();
-                    Duration offset = Duration.between( first, last )
-                                              .abs();
-                    offsets.add( offset );
-                }
+    /**
+     * Adds the time offset between adjacent time-series in the inputs.
+     * @param series the series
+     */
+    private Set<Duration> getOffsetsBetweenConsecutiveSeries( List<List<Event<T>>> series )
+    {
+        Set<Duration> offsets = new HashSet<>();
 
-                // Offset between the first and last series, which is required because the resampling is circular
-                Instant first = next.get( 0 )
-                                    .get( 0 )
-                                    .getTime();
-                List<Event<T>> lastSeries = next.get( next.size() - 1 );
-                Instant last;
+        // Offsets between consecutive series
+        for ( int i = 1; i < series.size(); i++ )
+        {
+            Instant first;
+            List<Event<T>> firstSeries = series.get( i - 1 );
 
+            if ( !firstSeries.isEmpty() )
+            {
                 if ( this.hasForecasts() )
                 {
-                    last = lastSeries.get( 0 )
-                                     .getTime();
+                    first = firstSeries.get( 0 )
+                                       .getTime();
                 }
                 else
                 {
-                    last = lastSeries.get( lastSeries.size() - 1 )
-                                     .getTime();
+                    first = firstSeries.get( firstSeries.size() - 1 )
+                                       .getTime();
                 }
 
+                Instant last = series.get( i )
+                                     .get( 0 )
+                                     .getTime();
                 Duration offset = Duration.between( first, last )
                                           .abs();
                 offsets.add( offset );
             }
         }
 
+        // Offset between the first and last series, which is required because the resampling is circular
+        Duration offset = this.getOffsetBetweenFirstAndLastSeries( series );
+        if ( Objects.nonNull( offset ) )
+        {
+            offsets.add( offset );
+        }
+
         return Collections.unmodifiableSet( offsets );
+    }
+
+    /**
+     * Gets the time offset between the first and last series.
+     * @param series the series
+     */
+    private Duration getOffsetBetweenFirstAndLastSeries( List<List<Event<T>>> series )
+    {
+        // Offset between the first and last series, which is required because the resampling is circular
+        List<Event<T>> firstSeries = series.get( 0 );
+        List<Event<T>> lastSeries = series.get( series.size() - 1 );
+
+        Duration offset = null;
+
+        if ( !firstSeries.isEmpty()
+             && !lastSeries.isEmpty() )
+        {
+            Instant first = firstSeries
+                    .get( 0 )
+                    .getTime();
+
+            Instant last;
+
+            if ( this.hasForecasts() )
+            {
+                last = lastSeries.get( 0 )
+                                 .getTime();
+            }
+            else
+            {
+                last = lastSeries.get( lastSeries.size() - 1 )
+                                 .getTime();
+            }
+
+            offset = Duration.between( first, last )
+                             .abs();
+        }
+
+        return offset;
     }
 
     /**
@@ -218,8 +259,12 @@ class BootstrapPool<T>
         List<List<TimeSeries<T>>> groupedBySize = new ArrayList<>();
         for ( Map.Entry<Integer, List<TimeSeries<T>>> nextEntry : bySize.entrySet() )
         {
+            int count = nextEntry.getKey();
             List<TimeSeries<T>> nextList = nextEntry.getValue();
-            nextList.sort( sorter );
+            if ( count > 0 )
+            {
+                nextList.sort( sorter );
+            }
             groupedBySize.add( nextList );
         }
 
@@ -237,7 +282,10 @@ class BootstrapPool<T>
                 List<TimeSeries<T>> nextInner = nextInnerEntry.getValue();
 
                 // Sort by the first valid time in each series
-                nextInner.sort( sorter );
+                if ( nextCount > 0 )
+                {
+                    nextInner.sort( sorter );
+                }
 
                 List<List<Event<T>>> unwrapped = nextInner.stream()
                                                           .map( n -> List.copyOf( n.getEvents() ) )

